### PR TITLE
Add some doc regarding the new features in the maven plugin

### PR DESCRIPTION
**Pull request by essobedo**
follow up of #16600

## Motivation

Several new features have been added into the maven plugin but as they are not documented, they don't exist so let's add some doc about them.

## Modifications:

* Describes shortly the features added by #15356, #16054 and #16600 in the Guide dedicated to maven

### CODE REVIEW

**Comment by essobedo** [comment]
@aloubyansky As promised, here is the doc. Please let me know if it is good enough or you need more.

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 1]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+=== Building with a specific configuration profile
+
+Quarkus supports xref:config-reference.adoc#profiles[configuration profiles] in order to provide a specific configuration according to the target environment.
+
+The profile can be provided directly in the maven build's command thanks to the System property `quarkus.profile` with a command of type `./mvnw package -Dquarkus.profile=<profile-name-here>`,
+however it is also possible to specify the profile directly into the pom file of your project either using the properties of project, the system properties of the plugin or the properties of the plugin.
+
+In order of precedence (greater precedence first):
+
+System properties of the plugin:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:580]:
```suggestion
1. System properties set in the Quarkus Maven plugin configuration
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 2]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+     </plugin>
+     ...
+    </plugins>
+  </build>
+...
+</project>
+----
+<1> The default configuration profile of this project is `prod-aws`.
+
+Properties of the plugin:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:609]:
```suggestion
2. Quarkus Maven plugin configuration properties
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 3]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+     </plugin>
+     ...
+    </plugins>
+  </build>
+...
+</project>
+----
+<1> The default configuration profile of this project is `prod-aws`.
+
+Properties of the project:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:638]:
```suggestion
3. Project properties
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 4]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+  <properties>
+    <quarkus.profile>prod-aws</quarkus.profile> <1>
+    ...
+  </properties>
+...
+</project>
+----
+<1> The default configuration profile of this project is `prod-aws`.
+
+NOTE: Whatever the approach chosen, it can still be overridden with the System property `quarkus.profile` or the environment variable `QUARKUS_PROFILE`.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:654]:
```suggestion
NOTE: Whatever the approach is chosen, the profile can still be overridden with the `quarkus.profile` system property or the `QUARKUS_PROFILE` environment variable.
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 5]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+
+[[maven-multi-build]]
+=== Building several artifacts from a single module
+
+In some particular use cases, it can be interesting to build several artifacts of your application from the same module.
+A typical example is when you want to build your application with different configuration profiles.
+
+In that case, it is possible to add as many executions as needed to the declaration of the Quarkus maven plugin like in the following example.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:662]:
```suggestion
In that case, it is possible to add as many executions as needed to the Quarkus Maven plugin configuration.
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 6]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+
+Example where the plugin will build the application twice one for Oracle and the other for PostgreSQL:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:664]:
```suggestion
Below is an example of a Quarkus Maven plugin configuration that will produce two builds of the same application: one  using the `prod-oracle` profile and the other one using the `prod-postgresql` profile.
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 7]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+  </build>
+...
+</project>
+----
+<1> The default configuration profile of the first execution of the plugin is `prod-oracle`.
+<2> The output directory of the first execution of the plugin is set to `oracle-quarkus-app` instead of `quarkus-app` to have a dedicated directory.
+<3> The default configuration profile of the second execution of the plugin is `prod-postgresql`.
+<4> The output directory of the second execution of the plugin is set to `postgresql-quarkus-app` instead of `quarkus-app` to have a dedicated directory.
+
+Since the plugin builds the application with the same dependencies, we may end up with dependencies that are not needed according to the configuration profile used.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:717]:
```suggestion
NOTE: With the configuration above, both profile builds will be using the same dependencies, so if we added dependencies on the Oracle and PostgreSQL drivers to the application, both of the drivers will appear in both builds.
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 8]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+
+Indeed, in our previous example, we would like to have the JDBC driver for Oracle only in the final artifact generated with the configuration profile `prod-oracle` and the same thing for PostgreSQL.
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:719]:
```suggestion
```

File: docs/src/main/asciidoc/maven-tooling.adoc [hunk 9]
```diff
@@ -567,6 +567,209 @@ unless it is the main application module already configured with the quarkus-mav
...
+
+By default, the final artifacts will contain both divers which is not what we want, to avoid that you can define both JDBC drivers as optional dependencies then indicate Quarkus that the optional dependencies need to be filtered and how as next:
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/maven-tooling.adoc:721]:
```suggestion
To isolate profile-specific dependencies from other profiles, the JDBC drivers could be added as optional dependencies to the application but configured to be included in each profile that requires them, e.g.:
```

**Review by aloubyansky — COMMENTED** [review]
Thanks a lot @essobedo! I added my adjustments. Could you please check whether they make sense or I have gone too far :)

**Comment by essobedo** [comment]
@aloubyansky much better thank you 🙏 I definitively need to learn how to write a good doc 😅

**Review by aloubyansky — APPROVED** [review]

**Pull request merged**